src/mobilenet.jl

```diff
@@ -0,0 +1,300 @@
+    invres = Chain(conv_bn((1, 1), inplanes, hidden_planes, relu6; bias = false)...,
+                   conv_bn((3, 3), hidden_planes, hidden_planes, relu6;
+                           bias = false, stride, pad = 1, groups = hidden_planes)...,
+                   conv_bn((1, 1), hidden_planes, outplanes, identity; bias = false)...)
+  end
+
+  (stride == 1 && inplanes == outplanes) ? SkipConnection(invres, +) : invres
+end
+
+"""
+    mobilenetv2(width_mult, configs; max_width = 1280, nclasses = 1000)
+
+Create a MobileNetv2 model.
+([reference](https://arxiv.org/abs/1801.04381)).
+
+# Arguments
+- `width_mult`: Controls the number of feature maps in each layer, with 1.0 being the original
+  model as detailed in the paper.
+- `configs`: A "list of lists" configuration for each layer that details:
+  - `t`: The expansion factor that controls the number of feature maps in the bottleneck layer
+  - `c`: The number of output feature maps
+  - `n`: The number of times a block is repeated
+  - `s`: The stride of the convolutional kernel
+- `max_width`: The maximum number of feature maps in any layer of the network
+- `nclasses`: The number of output classes
+"""
+function mobilenetv2(width_mult, configs; max_width = 1280, nclasses = 1000)
+  
+  # building first layer
+  inplanes = _make_divisible(32 * width_mult, width_mult == 0.1 ? 4 : 8)
+  layers = []
+  append!(layers, conv_bn((3, 3), 3, inplanes, stride = 2))
+
+  # building inverted residual blocks
+  for (t, c, n, s) in configs
+    outplanes = _make_divisible(c * width_mult, width_mult == 0.1 ? 4 : 8)
+    for i in 1:n
+      push!(layers, invertedresidualv2(inplanes, outplanes, i == 1 ? s : 1, t))
+      inplanes = outplanes
+    end
+  end
+
+  # building last several layers
+  outplanes = (width_mult > 1.0) ? _make_divisible(max_width * width_mult, width_mult == 0.1 ? 4 : 8) : max_width
+
+  return Chain(Chain(layers...,
+                     conv_bn((1, 1), inplanes, outplanes, relu6, bias = false)...),
+               Chain(AdaptiveMeanPool((1, 1)), flatten, Dense(outplanes, nclasses)))
+end
+
+# Layer configurations for MobileNetv2
+mobilenetv2_configs = [
+  # t, c, n, s
+  [1, 16, 1, 1],
+  [6, 24, 2, 2],
+  [6, 32, 3, 2],
+  [6, 64, 4, 2],
+  [6, 96, 3, 1],
+  [6, 160, 3, 2],
+  [6, 320, 1, 1],
+]
+
+# Model definition for MobileNetv2
+struct MobileNetv2
+  layers
+end
+
+"""
+    MobileNetv2(width_mult::Float64 = 1.0; pretrain = false, nclasses = 1000)
+
+Create a MobileNetv2 model with the specified configuration.
+([reference](https://arxiv.org/abs/1801.04381)).
+Set `pretrain` to `true` to load the pretrained weights for ImageNet. 
+
+# Arguments
+- `width_mult`: Controls the number of feature maps in each layer, with 1.0 being the original
+  model as detailed in the paper.
+- `pretrain`: Whether to load the pre-trained weights for ImageNet
+- `nclasses`: The number of output classes
+"""
+function MobileNetv2(width_mult::Float64 = 1.0; pretrain = false, nclasses = 1000)
```

Comment:
Does this have to be a `Float64`?

```suggestion
function MobileNetv2(width_mult = 1.0; pretrain = false, nclasses = 1000)
```

Reply:
Nope, realised that the function call works just fine without. Fixed